crates/cli/src/utils/mod.rs

```diff
@@ -110,6 +110,10 @@ pub fn get_provider_builder(config: &Config) -> Result<ProviderBuilder> {
         builder = builder.timeout(Duration::from_secs(rpc_timeout));
     }
 
+    if let Some(rpc_headers) = &config.eth_rpc_headers {
+        builder = builder.headers(rpc_headers.clone());
```

Comment:
Here we can check for `starts_with("User-Agent:")`, and if none we should add our own default user-agent